### DISCUSSION TITLE
Why is ios-src required for webxr?

### DISCUSSION THREAD

**Discussion by iiLearner** [google/model-viewer#4143] [Q&A]
Hi,

Forgive my little knowledge about the subject but as far as i know for webxr a threejs scene is rendered into immersive-ar, if so, isnt glb basically enough for ios and android both? why do we need an ios-src (usdz?) for webxr?

**Comment by elalish** [comment]
You don't - does it say this anywhere in our documentation? You don't even need an ios-src at all, even for iOS AR, since we now support auto-generation.

**Comment by iiLearner** [comment]
mhm, ive always used ios-src since i used scene viewer mostly however im switching to webxr so my initial thought was ios-src wouldnt be required.

when i dont give it ios-src, the ar button does not appear in ios even when using webxr.

ss: https://ilearner.dev/imgs/jnfof.png

**Reply by elalish** [reply]
`ios-src` is used by neither webxr nor scene-viewer, but only quick-look. You are not using webxr because webxr does not exist (yet) on iOS. You have specified `ar-modes='webxr scene-viewer'`, which is specifically disallowing quick-look, which is the only AR mode on iOS. If you remove both `ios-src` and `ar-modes` and just use our defaults, it should work just fine.
